cr-examples/onnx/src/main/java/oracle/code/onnx/provider/CoreMLProvider.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package oracle.code.onnx.provider;
+
+import oracle.code.onnx.OnnxRuntime;
+
+import java.util.logging.Logger;
+import java.util.logging.Level;
+
+import static oracle.code.onnx.foreign.coreml_provider_factory_h.*;
+
+public final class CoreMLProvider implements OnnxProvider {
+    private static final Logger logger = Logger.getLogger(CoreMLProvider.class.getName());
+
+    @Override
+    public void configure(OnnxRuntime.SessionOptions sessionOptions) {
+        var sessionOptionsAddress = sessionOptions.getSessionOptionsAddress();
+
+        try {
+            int coremlFlag = COREML_FLAG_USE_CPU_AND_GPU();
+            var status = OrtSessionOptionsAppendExecutionProvider_CoreML(sessionOptionsAddress, coremlFlag);
```

Comment:
The [ML_PROGRAM flag](https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/providers/coreml/coreml_provider_factory.h#L32) is probably useful to turn on here as it expands the set of things that ORT can accelerate with CoreML (e.g. I can't get it to do anything to [sd4j](https://github.com/oracle/sd4j) without it, but it does help a bit with it, or at least it did when I tested it in 2023). ML programs are the more modern way of accelerating things with CoreML, but they might need slightly higher OS versions.